src/crypto/base64.ts

```diff
@@ -14,8 +14,8 @@ class Base64 {
     return Utf8.stringify(CryptoBase64.parse(data));
   }
 
-  public static decodeJSON<T>(data: string): T {
-    return JSON.parse(Base64.decode(data));
+  public static decodeJSON<T>(data: string | undefined): T {
+    return JSON.parse(data ? Base64.decode(data) : '');
```

Comment:
i think `JSON.parse` of an empty string would throw an error. We could change that to `return data ? JSON.parse( Base64.decode(data)):  '';`

Reply:
that's the case, we should throw error or return the correct json

Reply:
otherwise, the code below will have runtime errors:

```
const data = base64.decodeJSON<{id: number}>(undefined);

console.log(data.id)
```